app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  include ActionView::Helpers::TextHelper
+  include InputSanitizeHelper
+  include ApplicationHelper
+  include CommentHelper
+
+  before_action :load_object, only: %i(index create)
+  before_action :load_comment, only: %i(update destroy)
+  before_action :check_view_permissions, only: :index
+  before_action :check_create_permissions, only: :create
+  before_action :check_manage_permissions, only: %i(update destroy)
+
+  def index
+    comments = @commentable.comments
+
+    render json: {
+      object_name: @commentable.name,
+      comments: render_to_string(partial: 'shared/comments/comments_list.html.erb',
+                                 locals: { comments: comments })
+    }
+  end
+
+  def create
+    @comment = @commentable.comments.new(
+      message: params[:message],
+      user: current_user,
+      associated_id: @commentable.id
+    )
+    comment_create_helper(@comment, 'comment')
```

Comment:
This structure is a bit weird for me, without response, render function.  
I would prefer to see here a service call that creates a comment, return comment, and explicit render here based on the result. 

That service could be reused later in API controllers.